sql/core/src/test/scala/org/apache/spark/sql/DataFrameComplexTypeSuite.scala

```diff
@@ -26,6 +27,55 @@ import org.apache.spark.sql.test.SharedSQLContext
 class DataFrameComplexTypeSuite extends QueryTest with SharedSQLContext {
   import testImplicits._
 
+  test("create an array") {
+    val df = sparkContext.parallelize(Seq(1, 2), 1).toDF("v")
+    df.selectExpr("Array(v + 3, v + 4)").collect
```

Comment:
I think we already have tests for array creation (if not, then the test coverage of spark is too bad...)